src/main/java/com/google/api/codegen/java/JavaDocConfig.java

```diff
@@ -68,101 +52,34 @@ public boolean isUnpagedListCallableVariant() {
-        JavaGapicContext context, Interface service, Method method) {
-      Map<String, Object> initFieldStructure = createInitFieldStructure(context, service, method);
-      InitCodeGenerator generator = new InitCodeGenerator();
-      InitCode initCode = generator.generateRequestObjectInitCode(method, initFieldStructure);
-      setInitCode(initCode);
-      return this;
-    }
+    public abstract Builder setParams(ImmutableList<InputParameter> params);
 
-    public JavaDocConfig.Builder setFieldInitCode(
-        JavaGapicContext context, Interface service, Method method, Iterable<Field> fields) {
-      Map<String, Object> initFieldStructure = createInitFieldStructure(context, service, method);
-      InitCodeGenerator generator = new InitCodeGenerator();
-      InitCode initCode =
-          generator.generateRequestFieldInitCode(method, initFieldStructure, fields);
-      setInitCode(initCode);
-      return this;
-    }
+    public abstract Builder setPagedVariant(boolean paged);
 
-    private static Map<String, Object> createInitFieldStructure(
-        JavaGapicContext context, Interface service, Method method) {
-      MethodConfig methodConfig =
-          context.getApiConfig().getInterfaceConfig(service).getMethodConfig(method);
-      Map<String, String> fieldNamePatterns = methodConfig.getFieldNamePatterns();
-
-      ImmutableMap.Builder<String, InitValueConfig> initValueConfigMap = ImmutableMap.builder();
-      for (Map.Entry<String, String> fieldNamePattern : fieldNamePatterns.entrySet()) {
-        CollectionConfig collectionConfig =
-            context.getCollectionConfig(service, fieldNamePattern.getValue());
-        InitValueConfig initValueConfig =
-            InitValueConfig.create(context.getApiWrapperName(service), collectionConfig);
-        initValueConfigMap.put(fieldNamePattern.getKey(), initValueConfig);
-      }
-      Map<String, Object> initFieldStructure =
-          FieldStructureParser.parseFields(
-              methodConfig.getSampleCodeInitFields(), initValueConfigMap.build());
-      return initFieldStructure;
-    }
+    public abstract Builder setCallableVariant(boolean callable);
 
-    public abstract JavaDocConfig.Builder setParams(ImmutableList<InputParameter> params);
-
-    public JavaDocConfig.Builder setRequestObjectParam(
-        JavaGapicContext context, Interface service, Method method) {
-      InputParameter param =
-          InputParameter.newBuilder()
-              .setType(method.getInputType())
-              .setName(REQUEST_PARAM_NAME)
-              .setDescription(REQUEST_PARAM_DOC)
-              .build();
-      return setParams(ImmutableList.of(param));
-    }
+    public abstract Builder setResourcesFieldForUnpagedListCallable(Field field);
 
-    public JavaDocConfig.Builder setEmptyParams() {
-      return setParams(ImmutableList.<InputParameter>of());
+    @Override
+    protected Builder setInitCodeProxy(InitCode initCode) {
+      return setInitCode(initCode);
```

Comment:
Possibly, but I had trouble getting it to work without them. I tried declaring methods like `public abstract setInitCode()` in the DocConfig super-class, but autovalue was not picking them up and defining them in the @AutoValue annotated sub-classes. I spent about an hour trying to figure it out (and reading through the autovalue docs) before doing the proxy methods and moving on to fixing other issues.

If someone has a better idea for how to do this, let's fix it in a subsequent PR.